src/lib/detect.py

```diff
@@ -0,0 +1,74 @@
+import os
+import subprocess
+from git import Repo, exc
+
+target_type = "path"
+
+def is_git_repo(target):
+    """
+    ## Check if a directory is a git repository.
+    ### Args:
+        target (str): The path to the directory.
+    """
+    try:
+        _ = Repo(target).git_dir
+        return True
+    except exc.InvalidGitRepositoryError:
+        return False
+
+def get_repo_name(target):
+    """
+    ## Get the name of a git repository.
+    ### Args:
+        target (str): The path to the git repository.
+    """
+    origin = subprocess.check_output(["git", "-C", target, "config", "--get", "remote.origin.url"], text=True).strip()
+    name = origin.split("/")[-1].replace(".git", "")
+    return name
+
+def get_repo_version(target):
+    """
+    ## Get the version of a git repository.
+    ### Args:
+        target (str): The path to the git repository.
+    """
+    version = subprocess.check_output(["git", "-C", target, "describe", "--tags"], text=True).strip()
+    return version
+
+def detect_target_type(target):
+    """
+    ## Detect the type of target. ISO or directory.
+    ### Args:
+        target (str): The path to the target.
+    """
+    global target_type
+    if target.endswith(".iso"):
+        target_type = "iso"
+        print("Detected iso file.")
+        name = os.environ.get("NAME")
+        if name is None or name == "None":
+            name = os.path.basename(target)
+        version = os.environ.get("VERSION")            
+        if version is not None and os.environ.get("VERSION") != "None":
+            version = os.environ.get("VERSION")
+        if version is None:
+            version = "undefined"
+        else:
+            version = version
+    elif os.path.isdir(target):
+        target_type = "directory"
+        print("Detected directory.")
+        if is_git_repo(target):
+            version = get_repo_version(target)
+            name = get_repo_name(target)
+            target_type = "git"
+            print(f"Detected git repository: {name}, version: {version}")
+        else:
+            name = os.path.basename(target)
+        if os.environ.get("VERSION") is not None and os.environ.get("VERSION") != "None":
+            version = os.environ.get("VERSION")
+            if version is None:
+                version = "undefined"
+            else:
+                version = version
```

Comment:
I'm not sure it's what you want here, you want the environment variable to potentially override the git repo version ?

If yes then you should do this
```suggestion
        version = os.environ.get("VERSION", "None")
        if version == "None":
            version = "undefined"
```

If not you have to change a bit the function to set `version = None` at the beginning and do something like this

```suggestion
        if version is None:
            version = os.environ.get("VERSION", "None")
            if version == "None":
                version = "undefined"
```